lib/services/cloud-project-service.ts

```diff
@@ -87,26 +86,40 @@ export class CloudProjectService extends CloudService implements ICloudProjectSe
 
 		const tasksResults: IDictionary<IServerResult> = {};
 		this.$logger.info(`${EOL}### Build machines cleanup${EOL}`);
-		for (let i = 0; i < cleanupResponse.buildMachineResponse.length; i++) {
-			const taskId = `Cloud Cleanup Task #${i + 1}`;
-			const task = cleanupResponse.buildMachineResponse[i];
-			try {
-				await this.waitForServerOperationToFinish(taskId, task);
-				const taskResult = await this.getObjectFromS3File<IServerResult>(task.resultUrl);
-				const output = this.$nsCloudOutputFilter.filter(await this.getContentOfS3File(task.outputUrl));
-				if (this.hasContent(output)) {
-					this.$logger.info(output);
+		const childTasks: Promise<IChildTaskExecutionResult>[] = [];
+		for (let task of cleanupResponse.buildMachineResponse) {
+			const executeChildTask = async (): Promise<IChildTaskExecutionResult> => {
```

Comment:
No need to define the function every time - you can extract it as a service method, e.g. waitForChildTaskToFinish